mm2src/coins/eth.rs

```diff
@@ -1306,9 +1305,12 @@ impl SwapOps for EthCoin {
     #[inline]
     fn derive_htlc_key_pair(&self, _swap_unique_data: &[u8]) -> keys::KeyPair {
         match self.priv_key_policy {
-            EthPrivKeyPolicy::KeyPair(ref key_pair) => {
-                key_pair_from_secret(key_pair.secret().as_bytes()).expect("valid key")
-            },
+            EthPrivKeyPolicy::Iguana(ref key_pair)
+            | EthPrivKeyPolicy::HDWallet {
+                activated_key: ref key_pair,
+                ..
+            } => key_pair_from_secret(key_pair.secret().as_bytes()).expect("valid key"),
+            EthPrivKeyPolicy::Trezor => todo!(),
```

Comment:
It's better to implement it panic free

Reply:
> if it's not due to some reason prefer panic with a clear message reasoning this panic

`todo!()` panics with "not yet implemented" message. This part of the code will never be executed since it's used for swaps and ETH will not be enabled for Trezor (which is a prerequisite for doing swaps) as it returns an error in such case.